### PR TITLE
Fixes Epinephrine Healing

### LINKED ISSUE #78474: Epinephrine no longer heals properly
Reporting client version: 514.1589

## Round ID:
[214999](https://scrubby.melonmesa.com/round/214999)

## Testmerges:
- [[Fuck Shotties] Old Yellers Old Yeller (Improvised Shell Nerf)](https://www.github.com/tgstation/tgstation/pull/78235)

## Reproduction:

Spawn a human
Build mode throw them against a wall a bit or whatever
image
Add 25u epinephrine to them and come back 10u later
image
When the epinephrine runs out they will "update" and recover their full health, however the gradual progressive improvement of health will not be reflected on the character's mob ; e.g. while here we see total 86 brute the overall status is -14% (incorrectly) and the player has a flashing health bar, is horizontal, and crawls, as if in soft crit.

### PR DESCRIPTION
**Pull request by GPeckman**
## About The Pull Request

As described in #78474, epinephrine wasn't properly updating health when it healed people in crit. This meant that people stayed in crit despite not being damaged enough to reach the crit threshold (at least until something else updated their health). This fixes that. Closes #78474.
## Why It's Good For The Game

Epinephrine should actually get people out of crit.
## Changelog
:cl:
fix: Epinephrine will now update health properly.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
How in the world does returning ..() make a difference here

**Comment by GPeckman** [comment]
> How in the world does returning ..() make a difference here

Basically, some `metabolize` proc calls the `on_mob_life` proc for every relevant reagent. If the return value from `on_mob_life` is true, then `metabolize` will call `updatehealth` (which is good). The issues is that `return ..()` will return whatever the parent returns, or nothing if the parent returns nothing. In this case, the parent returns nothing, so `on_mob_life` returns nothing either, and health isn't updated.

On the other hand `..()` will not immediately return the function. You may notice how there's an earlier line in `on_mob_life`: `. = TRUE`. This sets the default return value, so without the explicit return statement, `TRUE` is what gets returned. This means that health IS updated, fixing the bug.

**Comment by carlarctg** [comment]
sounds like a define would be useful. For readability 

#define RETURN_UPDATE_HEALTH

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -877,7 +876,7 @@
 	affected_mob.adjustStaminaLoss(-0.5 * REM * seconds_per_tick, 0)
 	if(SPT_PROB(10, seconds_per_tick))
 		affected_mob.AdjustAllImmobility(-20)
-	return ..()
```
Comment (SyncIt21) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:880]: Don't remove this. Yes we don't need the return value but you should still call the parent proc cause it does some important stuff 

https://github.com/tgstation/tgstation/blob/717e0147c18396d12b995354ee18ea4ef36a138f/code/modules/reagents/chemistry/reagents/medicine_reagents.dm#L12-L16

Comment (GPeckman) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:880]: Alright, now it has `..()` on one line and `return TRUE` on the next.

**Comment by ExcessiveUseOfCobblestone** [comment]
why doesnt parent return true too, isnt that the real fix since thats the expected default behavior?

**Comment by GPeckman** [comment]
Its not the default behavior, exactly. The `metabolize_reagent` proc calls the procs `overdose_start`, `overdose_process`, `on_mob_life`, and `on_mob_dead`. If any of those return true, then it sets the mob's health to be updated. Not all chems do stuff related to health, though, so returning true isn't the default behavior.

Carlarc is probably right about a define being good here, but that also sounds like a lot of work.

**Pull request merged**